src/main/java/ch/wisv/areafiftylan/products/service/OrderServiceImpl.java

```diff
@@ -168,16 +185,21 @@ public Order updateOrderStatus(String orderReference) {
     }
 
     private void validateTicketsIfPaid(Order order) {
+        if (order.getUser() == null) {
+            throw new UnassignedOrderException(order.getId());
+        }
+
         if (order.getStatus().equals(OrderStatus.PAID)) {
             for (Ticket ticket : order.getTickets()) {
+                ticketService.assignTicketToUser(ticket.getId(), order.getUser().getUsername());
```

Comment:
As this method is called at least once per paid order, you're sure this happens once for each ticket. When a ticket is invalid, it doesn't really need a user anyway.